### ISSUE TITLE
Chunk Destroyer not outputting items until power is removed

### ISSUE THREAD

**Issue by iiTranquility** [Kotori316/QuarryPlus#337]
## Describe the bug

QuarryPlus Chunk Destroyer doesn't output large amount of items until power is turned off.

## To Reproduce

1. Have a chunk destroyer mine a 40x40 area.
My settings are
Left: 40
Top: 40
Right: 40
Bottom: -1

2. Have a dank/null inside of a dank dock. 
3. Put dank dock down next to chunk destroyer
4. Put dank/null inside of dank dock
5. Give Chunk Destroyer Power
6. Use Status Checker (once mining) to see "items in total" going up while still out putting items into the dank/null via dock.
7. Turn off/remove power from Chunk Destroyer and all items will be exported from chunk destroyer almost instantly

If your server/client has crashed, please give me the crash report.

## Versions

* Minecraft version: [1.16.5]
* Single Player or Multi Player?: [Single player LAN hosted]
* Forge version: [1.16.5]
* QuarryPlus version: [16.1.11]
* Other mods: [Everything in ATM 6]

## Crash report or logs

If your Minecraft has crashed, please attach the crash report. Please use pastebin or just paste the report.

```
#Beginning of QuarryPlus configuration.
[common]
	#True to enable debug mode.
	debug = true
	#True to make machines work with no energy.
	NoEnergy = false
	#Amount of energy WorkbenchPlus can receive in a tick. [MJ]
	#Range: > 0
	WorkbenchMaxReceive = 250
	#The amount of energy(FE) workbench receives in a tick.
	#Range: > 0
	WorkbenchTickBonus = 5
	#Fasten quarry's head moving.
	FastQuarryHeadMove = false
	#True to enable ChunkDestroyer to collect bedrock as item if the machine has Bedrock Module in its module slots.
	CollectBedrock = true
	#DisableFrameChainBreak
	DisableFrameChainBreak = false
	#Set false to allow PumpPlus to remove non-source fluid block.
	RemoveOnlyFluidSource = false
	#True to enable RS control of machines.
	EnableRSControl = false
	#Range limit of ChunkDestroyer. set -1 to disable. The unit of number is `blocks`. 16 = 1 chunk.
	#Range: > -1
	QuarryRangeLimit = -1
	#Spawner Controller Blacklist
	spawnerBlacklist = ["minecraft:ender_dragon", "minecraft:wither", "minecraft:area_effect_cloud", "minecraft:item", "minecraft:player"]

	#Quarry PowerSetting [MJ] (min = 0, Max = 2,000,000,000 = 2 billion)
	[common.PowerSetting]

		[common.PowerSetting.Quarry]

			[common.PowerSetting.Quarry.BreakBlock]
				#Range: 0.0 ~ 2.0E9
				BasePower = 40.0
				#Range: 0.0 ~ 2.0E9
				EfficiencyCoefficient = 1.3
				#Range: 0.0 ~ 2.0E9
				UnbreakingCoefficient = 1.0
				#Range: 0.0 ~ 2.0E9
				FortuneCoefficient = 1.3
				#Range: 0.0 ~ 2.0E9
				SilktouchCoefficient = 2.0
				#Range: 0.0 ~ 2.0E9
				BaseMaxReceive = 300.0
				#Range: 0.0 ~ 2.0E9
				BaseMaxStored = 15000.0

				[common.PowerSetting.Quarry.BreakBlock.MoveHead]
					#Range: 0.0 ~ 2.0E9
					BasePower = 200.0
					#Range: 0.0 ~ 2.0E9
					UnbreakingCoefficient = 1.0

			[common.PowerSetting.Quarry.MakeFrame]
				#Range: 0.0 ~ 2.0E9
				BasePower = 25.0
				#Range: 0.0 ~ 2.0E9
				EfficiencyCoefficient = 1.3
				#Range: 0.0 ~ 2.0E9
				UnbreakingCoefficient = 1.0
				#Range: 0.0 ~ 2.0E9
				BaseMaxReceive = 100.0
				#Range: 0.0 ~ 2.0E9
				BaseMaxStored = 15000.0

		[common.PowerSetting.Pump]

			[common.PowerSetting.Pump.DrainLiquid]
				#Range: 0.0 ~ 2.0E9
				BasePower = 10.0
				#Range: 0.0 ~ 2.0E9
				UnbreakingCoefficient = 1.0

			[common.PowerSetting.Pump.MakeFrame]
				#Range: 0.0 ~ 2.0E9
				BasePower = 25.0
				#Range: 0.0 ~ 2.0E9
				UnbreakingCoefficient = 1.0

		[common.PowerSetting.MiningWell]
			#Range: 0.0 ~ 2.0E9
			BasePower = 40.0
			#Range: 0.0 ~ 2.0E9
			EfficiencyCoefficient = 1.3
			#Range: 0.0 ~ 2.0E9
			UnbreakingCoefficient = 1.0
			#Range: 0.0 ~ 2.0E9
			FortuneCoefficient = 1.3
			#Range: 0.0 ~ 2.0E9
			SilktouchCoefficient = 2.0
			#Range: 0.0 ~ 2.0E9
			BaseMaxReceive = 100.0
			#Range: 0.0 ~ 2.0E9
			BaseMaxStored = 1000.0

		[common.PowerSetting.Laser]
			#Range: 0.0 ~ 2.0E9
			BasePower = 4.0
			#Range: 0.0 ~ 2.0E9
			EfficiencyCoefficient = 2.0
			#Range: 0.0 ~ 2.0E9
			UnbreakingCoefficient = 0.1
			#Range: 0.0 ~ 2.0E9
			FortuneCoefficient = 1.05
			#Range: 0.0 ~ 2.0E9
			SilktouchCoefficient = 1.1
			#Range: 0.0 ~ 2.0E9
			BaseMaxReceive = 100.0
			#Range: 0.0 ~ 2.0E9
			BaseMaxStored = 1000.0

		[common.PowerSetting.Refinery]
			#Range: 0.0 ~ 2.0E9
			EfficiencyCoefficient = 1.2
			#Range: 0.0 ~ 2.0E9
			UnbreakingCoefficient = 1.0
			#Range: 0.0 ~ 2.0E9
			BaseMaxReceive = 6.0
			#Range: 0.0 ~ 2.0E9
			BaseMaxStored = 1000.0

	#Setting of enabling or disabling each machine. True to disable.
	[common.machines]
		ChunkDestroyer = false
		ExpPump = false
		SolidFuelQuarry = false
		Replacer = true
		NewQuarry = false
		MiniQuarry = false
		MarkerPlus = false
		Placer = false
		CreativeGenerator = false
		WorkbenchPlus = false
		QuarryPlus = false
		AdvancedPump = false
		EnchantMoverFromBook = true
		PumpPlus = false
		MiningwellPlus = false
		EnchantMover = false
		SpawnerController = true
		ModulePump = false
		ModuleExpPump = false
		ModuleReplacer = false
		ModuleTorch = false
		ModuleFuel = false
		ModuleRemoveBedrock = false

#Integration with other mods
[integration]
	#False to disable notification of chunk protection by FTBChunks
	SendNotificationOfChunkProtection = true
```

**Comment by Kotori316** [comment]
What "a dank/null inside of a dank dock"? What does it mean?

Use vanilla chest instead of "a dank dock"(?). If the same problem happens, tell me.

**Comment by iiTranquility** [comment]
https://i.iitranq.co.uk/z9PiAcn2Xm.mp4

This is my own personal image host, I uploaded it here instead of YT.

In the video, I show you the dank dock with the dank null inside of it, not outputting all of the items which is what the Status Checker shows.

I've done the same with the obsidian chest and a normal minecraft chest, so it isn't a storage mod issue.

> What "a dank/null inside of a dank dock"? What does it mean?
> 
> Use vanilla chest instead of "a dank dock"(?). If the same problem happens, tell me.

**Comment by iiTranquility** [comment]
It's a shame that it needs an inventory first, it's impossible to use Pipez pipes or Refined Storage importers.
Is there any way to enable this?

**Comment by Kotori316** [comment]
Thank you for the video. I understand the inventory.
Chunk Destroyer pushes just 1 type of item in a tick, ignoring the count. An example with Flint is that the machine mines the item and put to Chest in tick 1. In tick 2, if the machine mines Flint again then it puts Flint, leaving other items in machine inventory. When you stop supplying energy, the machine puts Flint, Wood, Ores, etc. in order as it never gets new items.

I changed to use a logic like FIFO(first in first out), so it might improve the behavior. The version is released as 16.1.15, available in [Curse](https://legacy.curseforge.com/minecraft/mc-mods/additional-enchanted-miner/files/5116853)(under review by moderator) and [Modrinth](https://modrinth.com/mod/additional-enchanted-miner/version/16.1.15).

> It's a shame that it needs an inventory first, it's impossible to use Pipez pipes or Refined Storage importers. Is there any way to enable this?

No. Quarry's storage is implemented to keep all mined items, which causes some problem when extracted from outside. (Not meaning Pipez do wrong extracting, but some mods cause something unexpected like item duplication) So I disable the function. It's intended.

For Refined Storage, I think you can use Interface to get items from quarry. Just place it next to quarry, then quarry puts items into RS Interface and RS network imports the items. I didn't test, but the same logic worked fine with AE2.

**Comment by iiTranquility** [comment]
https://i.iitranq.co.uk/ihxIciPv3q.mp4

That worked flawlessly, thank you so much for the fast fix :)

Closing now, appreciate you again!
